scripts/content/nzbindex.js

```diff
@@ -1,84 +1,102 @@
+    } else {
+        $(this).css('background-image', 'url(' + img + ')');
 
-	    //grab all checked boxes on page
-		var a = document.getElementsByTagName('input');
-		for (var i=0, len=a.length; i<len; ++i) {
-			if (a[i].type == 'checkbox' && a[i].checked) {
-				if ($(a[i]).parent().parent().find('td').length < 4) {
-					continue;
-				}
-				var categories = $(a[i]).parent().parent().find('td')[3].innerText;
-				if (categories.indexOf('\n') != -1) {
-					var category = categories.substr(0, categories.indexOf('\n'));
-				} else {
-					var category = categories;
-				}
-				if (use_nice_name_nzbindex == '1') {
-					var nice_name = $($(a[i]).parent().parent().find('td')[1]).find('label')[0].innerText;
-				}
-				addToSABnzbd(addLink, 'https://nzbindex.com/download/' + a[i].value + '/' + escape(nice_name), "addurl", nice_name, category);
-			}
-		}
-	}
+        //grab all checked boxes on page
+        var a = document.getElementsByTagName('input');
+        for (var i = 0, len = a.length; i < len; ++i) {
+            if (a[i].type == 'checkbox' && a[i].checked) {
+                if ($(a[i]).parent().parent().find('td').length < 4) {
+                    continue;
+                }
+                var categories = $(a[i]).parent().parent().find('td')[3].innerText;
+                if (categories.indexOf('\n') != -1) {
+                    var category = categories.substr(0, categories.indexOf('\n'));
+                } else {
+                    var category = categories;
+                }
+                if (use_nice_name_nzbindex == '1') {
+                    var nice_name = $($(a[i]).parent().parent().find('td')[1]).find('label')[0].innerText;
+                }
+                addToSABnzbd(addLink, 'https://nzbindex.com/download/' + a[i].value + '/' + escape(nice_name), "addurl", nice_name, category);
+            }
+        }
+    }
 
-	return false;
+    return false;
 }
 
 function handleAllDownloadLinks() {
-	$('input[value="Create NZB"]').each(function() {
-		// add button to send checked items to SABConnect
-		var img = chrome.extension.getURL('/images/sab2_16.png');
-		var link = '<input class="addSABnzbd" type="button" value="      Download selected" style="background-image: url('+img+'); background-repeat: no-repeat; background-position: 3px 1px;" />';
-		$(this).after(link);
-		$(this).parent().find('input[class="addSABnzbd"]').first().click(addToSABnzbdFromNzbindex);
-	});
-	
-	$('table a[href*="nzbindex.nl\\/download\\/"]').each(function() {
-	    var img = chrome.extension.getURL('/images/sab2_16.png');
-	    var href = $(this).attr('href');
-	    var link = '<a class="addSABnzbdOnClick" href="' + href + '"><img title="Send to SABnzbd" src="' + img + '" /></a> ';
-	    $(this).before(link);
-	});
+    $('input[value="Create NZB"]').each(function () {
+        if ($(this).attr('x-nzbpatched') !== 'true') {
+            $(this).attr('x-nzbpatched', 'true');
```

Comment:
Missing `x-nzbpatched="true"` on line 54?

Reply:
Sorry, I missed the notification for the comment.

The `x-nzbpatched` attribute was added because the old code was only working if that portion of code was run once. Because nzbindex now used async loading of result, it should be run each time the DOM is changed. It should not add a new link each time.

The lines like 50 and 51 should have been enough for that but... The pattern to find the "nzb download link" also trigger the new "add nzb link". Thus I've also added the `x-nzbpatched` attribute to the new created link.

The new input doesn't trigger the query used for addition, thus I didn't added it.

After second reflexion, it doesn't break anything, and in case of code evolution where the new added "input" would trigger the query, I've also added it in a new commit.

Also, there was a bug were the line 81 was adding several time the same trigger (instead of replacing it) at each run of the function, thus I've moved the code in the creation of the link.